### PR TITLE
Create ClassLabel for labelling tasks datasets

### PR DESCRIPTION
**Pull request by jplu**
This PR adds a specific `ClassLabel` for the datasets that are about a labelling task such as POS, NER or Chunking.

### CODE REVIEW

File: datasets/xtreme/xtreme.py [hunk 1]
```diff
@@ -450,12 +450,46 @@ def _info(self):
             features["label"] = datasets.Value("string")
         if self.config.name == "XNLI":
             features["gold_label"] = datasets.Value("string")
+        
+        if self.config.name.startswith("udpos"):
+            features = datasets.Features(
+                {
+                    "token": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/xtreme/xtreme.py:457]: `tokens` here ?

File: datasets/conll2000/conll2000.py [hunk 2]
```diff
@@ -77,9 +77,33 @@ def _info(self):
             features=datasets.Features(
                 {
                     "id": datasets.Value("string"),
-                    "words": datasets.Sequence(datasets.Value("string")),
-                    "pos": datasets.Sequence(datasets.Value("string")),
-                    "chunk": datasets.Sequence(datasets.Value("string")),
+                    "tokens": datasets.Sequence(datasets.Value("string")),
+                    "pos_tags": datasets.Sequence(datasets.Value("string")),
```
Comment (lhoestq) [inline datasets/conll2000/conll2000.py:81]: ClassLabel here ?

File: datasets/germeval_14/germeval_14.py [hunk 3]
```diff
@@ -85,8 +85,60 @@ def _info(self):
                     "id": datasets.Value("string"),
                     "source": datasets.Value("string"),
                     "tokens": datasets.Sequence(datasets.Value("string")),
-                    "labels": datasets.Sequence(datasets.Value("string")),
-                    "nested-labels": datasets.Sequence(datasets.Value("string")),
+                    "labels": datasets.Sequence(datasets.features.ClassLabel(names=[
```
Comment (lhoestq) [inline datasets/germeval_14/germeval_14.py]: ner_tags instead of labels ?

**Review by lhoestq — COMMENTED** [review]
cool !
So to summarize it seems to follow the convention:
- the tokens are in the `tokens` columns as sequence of str (note that sometimes the tokenization is only splitting words simply by space)
- the tags for POS, chunk and NER are in columns `pos_tags`, `chunk_tag` and `ner_tags` as sequence of ClassLabel

File: datasets/xtreme/xtreme.py (hunk 1 above)
Comment (jplu) [reply datasets/xtreme/xtreme.py:457]: No, here they are single token, not a sequence.

Comment (lhoestq) [reply datasets/xtreme/xtreme.py:457]: Yes right !

File: datasets/germeval_14/germeval_14.py [hunk 4]
```diff
@@ -85,8 +85,68 @@ def _info(self):
                     "id": datasets.Value("string"),
                     "source": datasets.Value("string"),
                     "tokens": datasets.Sequence(datasets.Value("string")),
-                    "labels": datasets.Sequence(datasets.Value("string")),
-                    "nested-labels": datasets.Sequence(datasets.Value("string")),
+                    "labels": datasets.Sequence(
```
Comment (stefan-it) [inline datasets/germeval_14/germeval_14.py:88]:
```suggestion
                    "ner_tags": datasets.Sequence(
```

File: datasets/germeval_14/germeval_14.py [hunk 5]
```diff
@@ -85,8 +85,68 @@ def _info(self):
...
+                                "B-PER"
+                                "I-PER"
+                                "B-PERderiv"
+                                "I-PERderiv"
+                                "B-PERpart"
+                                "I-PERpart"
+                            ]
+                        )
+                    ),
+                    "nested-labels": datasets.Sequence(
```
Comment (stefan-it) [inline datasets/germeval_14/germeval_14.py:119]:
```suggestion
                    "nested_ner_tags": datasets.Sequence(
```

Comment (stefan-it) [reply datasets/germeval_14/germeval_14.py:119]: Needs to be changed in line 150:

```python
yield sentence_counter, {
                "id": str(sentence_counter),
                "tokens": current_tokens,
                "ner_tags": current_labels,
                "nested_ner_tags": current_nested_labels,
                "source": current_source,
            }
```

**Comment by jplu** [comment]
@lhoestq Better?

**Review by lhoestq — APPROVED** [review]
awesome thanks :)
feel free to merge

**Pull request merged**